client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoPreparedStatement.java

```diff
@@ -123,7 +123,7 @@
     public void close()
             throws SQLException
     {
-        super.execute(format("DEALLOCATE PREPARE %s", statementName));
+        optionalConnection().ifPresent(x -> x.removePreparedStatement(statementName));
```

Comment:
maybe name the variable `connection`

Reply:
I did that originally, but I feel like it just makes the line longer (repeated twice) without improving readability.